### PR TITLE
Send DSN with SentryEnvelopeHeader

### PR DESCRIPTION
**Pull request by vishnukvmd**
## :scroll: Description
This change injects DSN into the `SentryEnvelopeHeader` class and serializes it if available.

## :bulb: Motivation and Context
This change ensures that the DSN is passed to the server along with events, bringing in consistency with the JavaScript SDK. This gets us one step closer to passing a `tunnel` that can be used to proxy data to Sentry servers (to fixe #872). Once this change is merged, will create a separate PR that accepts `tunnel` as a parameter within the `SentryConfig` and configures the transport accordingly (again, similar to the JS SDK).

## :green_heart: How did you test it?
Verified that the `dsn` was passed within the envelope to the remote server.

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by marandaneto** [comment]
@vishnukvmd Thanks for doing this.
That would require changes on the Android SDK as well since it does a roundtrip in the JSON, otherwise, this field is gonna be lost.
https://github.com/getsentry/sentry-java/blob/6.x.x/sentry/src/main/java/io/sentry/SentryEnvelopeHeader.java

Would you like to raise an issue on that repo as well? Thanks.

**Comment by vishnukvmd** [comment]
Hey @marandaneto, why would this require changes to the Java SDK? Isn't the Dart SDK self contained?

**Comment by marandaneto** [comment]
> Hey @marandaneto, why would this require changes to the Java SDK? Isn't the Dart SDK self contained?

Hey, no, the Android SDK has to deserialize and serialize again the event in order to enrich the event with the device context.
The reason is: https://github.com/getsentry/team-mobile/issues/11

On iOS it'd work ootb.

**Comment by vishnukvmd** [comment]
Yikes, I had no idea. I would much rather someone who is familiar with the development environment for the Java SDK make that change. I hope that's okay.

**Comment by marandaneto** [comment]
> Yikes, I had no idea. I would much rather someone who is familiar with the development environment for the Java SDK make that change. I hope that's okay.

Yep, let me try to get into this latest next week since this is a short week, if that's ok.

**Comment by vishnukvmd** [comment]
> > Yikes, I had no idea. I would much rather someone who is familiar with the development environment for the Java SDK make that change. I hope that's okay.
> 
> Yep, let me try to get into this latest next week since this is a short week, if that's ok.

That'd be great, thank you so much! 🙏

File: flutter/pubspec.yaml [hunk 1]
```diff
@@ -14,7 +14,8 @@ dependencies:
     sdk: flutter
   flutter_web_plugins:
     sdk: flutter
-  sentry: ^6.5.1
+  sentry:
+    path: ../dart
```
Comment (marandaneto) [inline flutter/pubspec.yaml]: This is not needed, there's a `dependency_overrides` already.

Comment (vishnukvmd) [reply flutter/pubspec.yaml]: Fixed, thank you!

**Comment by marandaneto** [comment]
@ua741 CI is unhappy, also, add a new changelog entry, after that, happy to LGTM.

**Comment by marandaneto** [comment]
> @ua741 CI is unhappy, also, add a new changelog entry, after that, happy to LGTM.

@ua741 any update here?

**Comment by krystofwoldrich** [comment]
Seems like only a lint issue, preferred const over final, otherwise tests are passing.

**Comment by marandaneto** [comment]
Closing in favor of https://github.com/getsentry/sentry-dart/pull/1050

**Comment by ua741** [comment]
> > @ua741 CI is unhappy, also, add a new changelog entry, after that, happy to LGTM.
> 
> @ua741 any update here?

Sorry for dropping the ball here, I was on vacation. 
Thank you so much for taking it forward and merging it. Really appreciate it.

**Pull request closed without merging**